### PR TITLE
1099 cli ingest cleanup cache

### PR DESCRIPTION
**Pull request by dai-dao**
### Description
add clean_cache argument to ingest cli

./quickwit index ingest --index hdfs-logs --clean_cache

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

**Comment by fulmicoton** [comment]
@PSeitz can you review?

File: quickwit-core/src/index.rs [hunk 1]
```diff
@@ -114,6 +116,28 @@ pub async fn delete_index(
     Ok(deleted_entries)
 }
 
+/// Cleans up split cache in local split store.
+///
+/// * `metastore_uri` - The metastore URI for accessing the metastore.
+/// * `index_id` - The target index Id.
+/// * `grace_period` -  Threshold period after which a staged split can be garbage collected.
+/// * `dry_run` - Should this only return a list of affected files without performing deletion.
```
Comment (PSeitz) [inline quickwit-core/src/index.rs:124]: these comments don't fit the function paramters

File: quickwit-core/src/index.rs [hunk 2]
```diff
@@ -114,6 +116,28 @@ pub async fn delete_index(
...
+pub async fn clean_split_cache(
+    data_dir_path: &Path,
+    index_id: String,
+    source_id: String,
+) -> anyhow::Result<()> {
+    let cache_path = data_dir_path
+        .join("indexing")
+        .join(source_id)
+        .join(index_id)
+        .join("cache");
```
Comment (PSeitz) [inline quickwit-core/src/index.rs:134]: "indexing" and "cache" are easy to break on changes, can you move them to variables and use them everywhere?

**Comment by PSeitz** [comment]
Thanks for the PR! I left some comments. Can you also add a test in quickwick-cli/tests/cli.rs?

**Comment by dai-dao** [comment]
thanks i added the changes

File: quickwit-cli/tests/cli.rs [hunk 3]
```diff
@@ -50,6 +53,21 @@ fn create_logs_index(test_env: &TestEnv) {
     .success();
 }
 
+fn ingest_docs_with_options(input_path: &Path, test_env: &TestEnv, options: &str) {
```
Comment (PSeitz) [inline quickwit-cli/tests/cli.rs:56]: can you let this function be called from `ingest_docs` and add the assertions from there

Comment (dai-dao) [reply quickwit-cli/tests/cli.rs:56]: you mean to call `ingest_docs_with_options` from `ingest_docs`? but `ingest_docs` doesn't take an extra `options` parameter to pass in the `--clean-cache`, unless you mean something else.

Comment (PSeitz) [reply quickwit-cli/tests/cli.rs:56]: I meant to unduplicate the code from `ingest_docs(input_path: &Path, test_env: &TestEnv` by calling `ingest_docs_with_options`

**Comment by PSeitz** [comment]
Looks good so far, but the tests are failing. 
I had a look, but couldn't see why the caching directory still exists in the test without the cleanup cache

**Comment by dai-dao** [comment]
yea i'm aware, will update

**Review by PSeitz — APPROVED** [review]

**Comment by PSeitz** [comment]
Looks good, thanks for the PR!

**Pull request merged**